exercise_004_extension_methods/README.md

```diff
@@ -4,10 +4,59 @@
 
 Extension methods can be used to add methods to types after they are defined.
 
-## Steps
+Lets say we need to add a method `square` to type `Int`. We used to achieve it 
+with the help of `implicit class`es in _Scala 2_
+
+```scala
+implicit class RichInt(val i: Int) extends AnyVal {
+  def square = i * i
+}
+```
+
+Now you can use it as follows
+
+```scala
+4.square
+// res1: Int = 16
+```
+
+With dotty `extension methods`, we can rewrite the above as
+
+```scala
+def (i: Int).square: Int = i * i
+
+3.square
```

Comment:
I would use the same number (4) instead of 3 as you specifically say _"we can rewrite the above as"_